### PR TITLE
Create custom components 

### PR DESCRIPTION
**Pull request by oreoluwade**
#### What does this PR do?
- Creates custom components for the Food-Basket application.

#### Description of Task to be completed?
- Create `Text`, `Button`, and `Box` components

#### How should this be manually tested?
- Check out to the `feature-create-custom-components-153117957` branch.
- Click on either the `signin` or `signup` tabs to see the components rendered in the modal.
- Check the `CustomComponents` directory for created components.

#### Any background context you want to provide?
- Creating components that can be used generally through the app will save a lot of work going forward. Due to the eclectic props each of these components have, no matter the styling framework being used, we can still use the components. The passing of props becomes the only thing necessary for a component to be properly styled anywhere in the app.

- Also, we can gradually scale our components to the extent that at some time in the future, we rely solely on our own components. 

#### What are the relevant pivotal tracker stories?
- [#153117957](https://www.pivotaltracker.com/story/show/153117957)

### CODE REVIEW

File: src/components/CustomComponents/Modal.js [hunk 1]
```diff
@@ -1,32 +1,37 @@
...
-      onClick={onClick}
-    />,
-    <FlatButton
-      key={2}
-      label="Submit"
-      primary
-      onClick={onClick}
-    />,
+    <Box key={1}>
+      <ActionButton
```
Comment (fisayomi) [inline src/components/CustomComponents/Modal.js]: We seem to now have duplicate buttons, One on the form(s), and another on the modal @oreoluwade

Comment (oreoluwade) [reply src/components/CustomComponents/Modal.js]: Yes, we seemed to. Updated now. @fisayomi

**Comment by oreoluwade** [comment]
@fisayomi Perhaps, the `Any background context you want to provide?` section of the PR description might provide more context. Let me know what you think please.

**Comment by fisayomi** [comment]
Okay, makes sense @oreoluwade [This](https://www.sitepoint.com/front-end-frameworks-custom-vs-ready-use-solutions/) gave me more insight.

**Comment by fisayomi** [comment]
Can you check the hound errors?

File: src/components/CustomComponents/Modal.js [hunk 2]
```diff
@@ -1,36 +1,39 @@
...
+
 const Modal = ({
   onClick,
   openModal,
   title,
   signup
 }) => {
   const actions = [
-    <FlatButton
+    <ActionButton
```
Comment (fisayomi) [inline src/components/CustomComponents/Modal.js]: We should probably put all buttons on the form. It looks somehow this way.

Comment (oreoluwade) [reply src/components/CustomComponents/Modal.js]: Yeah, I get that. However, as it is, the modal only closes if there is a button on it to trigger the close action. Should find a resolution/logic for that when it comes to ensuring that the signup functionality (not just the form) works properly.

**Pull request merged**